### PR TITLE
Add the possibility to trip a busbar section.

### PR DESCRIPTION
**Pull request by EstherDarkish**
Needs to update the OperatingStatus in Powsybl-core and add the BUSBAR_SECTION in the isAllowedIdentifiable function to test properly.
Ask me how to if the next version of powsybl is not used yet.

Edit : This is ready

### CODE REVIEW

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:2]: 2024

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        TestUtils.setOperatingStatus(network, TARGET_BUSBAR_ID, OTHER_BRANCH_STATUS);
+        return network;
+    }
+
+    @Override
+    protected ModificationInfos buildModification() {
+        return OperatingStatusModificationInfos.builder()
+                .stashed(false)
+                .equipmentId(TARGET_BUSBAR_ID)
+                .energizedVoltageLevelId("energizedVoltageLevelId")
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:48]: why do we need energizedVoltageLevelId here?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+import static com.powsybl.iidm.network.extensions.OperatingStatus.Status.FORCED_OUTAGE;
+import static com.powsybl.iidm.network.extensions.OperatingStatus.Status.PLANNED_OUTAGE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Tag("IntegrationTest")
+public class OperatingStatusModificationTripBusBarTest extends AbstractNetworkModificationTest {
+
+    private static final String TARGET_BUSBAR_ID = "1.A";
+    private static final String UPDATE_BRANCH_ID = "line1";
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:31]: yes but it's a line?
Why not use another BBS like "1B"?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    private static final OperatingStatus.Status TARGET_BRANCH_STATUS = FORCED_OUTAGE;
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:32]:
```suggestion
    private static final OperatingStatus.Status TARGET_BUSBAR_STATUS = FORCED_OUTAGE;
```

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    private static final OperatingStatus.Status OTHER_BRANCH_STATUS = PLANNED_OUTAGE;
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:33]:
```suggestion
    private static final OperatingStatus.Status OTHER_BUSBAR_STATUS = PLANNED_OUTAGE;
```

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                .energizedVoltageLevelId("energizedVoltageLevelId")
+                .action(OperatingStatusModificationInfos.ActionType.TRIP).build();
+    }
+
+    @Override
+    protected ModificationInfos buildModificationUpdate() {
+        return OperatingStatusModificationInfos.builder()
+                .stashed(false)
+                .equipmentId(UPDATE_BRANCH_ID)
+                .energizedVoltageLevelId("energizedVoltageLevelIdEdited")
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:57]: why do we need energizedVoltageLevelId here?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        assertEquals("TRIP", createdValues.get("action"));
+        assertEquals("1.A", createdValues.get("equipmentId"));
+    }
+
+    @Override
+    @SneakyThrows
+    protected void testUpdateModificationMessage(ModificationInfos modificationInfos) {
+        assertEquals("OPERATING_STATUS_MODIFICATION", modificationInfos.getMessageType());
+        Map<String, String> updatedValues = mapper.readValue(modificationInfos.getMessageValues(), new TypeReference<>() { });
+        assertEquals("energizedVoltageLevelIdEdited", updatedValues.get("energizedVoltageLevelId"));
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:89]: why do we need energizedVoltageLevelId here?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        TestUtils.assertOperatingStatus(getNetwork(), TARGET_BUSBAR_ID, OTHER_BRANCH_STATUS);
+        assertTerminalsStatusAfterNetworkModification(true);
+    }
+
+    @Override
+    @SneakyThrows
+    protected void testCreationModificationMessage(ModificationInfos modificationInfos) {
+        assertEquals("OPERATING_STATUS_MODIFICATION", modificationInfos.getMessageType());
+        Map<String, String> createdValues = mapper.readValue(modificationInfos.getMessageValues(), new TypeReference<>() { });
+        assertEquals("energizedVoltageLevelId", createdValues.get("energizedVoltageLevelId"));
```
Comment (antoinebhs) [inline src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java:79]: why do we need energizedVoltageLevelId here?

**Review by antoinebhs — COMMENTED** [review]
Can be improve to test better the change?

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java (hunk 1 above)
Comment (EstherDarkish) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java]: Thanks

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java (hunk 2 above)
Comment (EstherDarkish) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java]: Right, it's not needed

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java (hunk 6 above)
Comment (EstherDarkish) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java]: removed

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java (hunk 7 above)
Comment (EstherDarkish) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java]: removed

File: src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java (hunk 8 above)
Comment (EstherDarkish) [reply src/test/java/org/gridsuite/modification/server/modifications/OperatingStatusModificationTripBusBarTest.java]: removed

**Review by antoinebhs — APPROVED** [review]
Code OK
Test OK

**Pull request merged**